include/OpenSpaceToolkit/Astrodynamics/EventCondition/SemiMajorAxisCondition.hpp

```diff
@@ -0,0 +1,83 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_EventConditions_SemiMajorAxisCondition__
+#define __OpenSpaceToolkit_Astrodynamics_EventConditions_SemiMajorAxisCondition__
+
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+
+#include <OpenSpaceToolkit/Mathematics/Objects/Vector.hpp>
+
+#include <OpenSpaceToolkit/Physics/Units/Derived.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Length.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/EventCondition.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace eventcondition
+{
+
+using ostk::core::types::Real;
+using ostk::core::types::String;
+
+using ostk::math::obj::VectorXd;
+
+using ostk::physics::units::Derived;
+using ostk::physics::units::Length;
+
+using ostk::astro::EventCondition;
+
+/// @brief                      A semi major axis based event condition
+
+class SemiMajorAxisCondition : public EventCondition
```

Comment:
How much of this can be "templated" if we wanted to add "Inclination". Can we group those by type at least? like `AngleCondition`, or `LengthCondition`? Would that make sense?

Reply:
We could potentially add another parent class for OrbitalElementCondition or something.
But I don't see any common code. I'm trying to be computationally efficient here so we're only calculating the element we need, than generating a COE object.

Reply:
Sorry for being late to this convo 🙏. My vote here is to go with the thing that makes us have to change our API less down the road. 
This class make sense to me. :) 

Maybe down the road it makes sense to expose a factory function:
`EventCondition:ConditionFromName("Semi Major Axis")` 

Just to make these a bit easier to work with